src/parser.rs

```diff
@@ -176,28 +184,30 @@ pub struct AACAudioPacket<'a> {
   pub aac_data: &'a [u8],
 }
 
-pub fn aac_audio_packet(input: &[u8], size: usize) -> IResult<&[u8], AACAudioPacket<'_>> {
+pub fn aac_audio_packet(input: &[u8], size: usize) -> IResult<&[u8], AACAudioPacket> {
   if input.len() < size {
     return Err(Err::Incomplete(Needed::new(size)));
   }
 
   if size < 1 {
     return Err(Err::Incomplete(Needed::new(1)));
   }
-  let (_remaining, packet_type) = be_u8(input)?;
-  let packet_type = match packet_type {
-    0 => AACPacketType::SequenceHeader,
-    1 => AACPacketType::Raw,
-    _ => return Err(Err::Error(Error::new(input, ErrorKind::Alt))),
-  };
-
-  Ok((
-    &input[size..],
-    AACAudioPacket {
-      packet_type,
-      aac_data: &input[1..size],
-    },
-  ))
+
+  be_u8(input).and_then(|(_, packet_type)| {
+    let packet_type = match packet_type {
+      0 => AACPacketType::SequenceHeader,
+      1 => AACPacketType::Raw,
+      _ => return Err(Err::Error(Error::new(input, ErrorKind::Alt))),
+    };
+
+    Ok((
+      &input[size..],
+      AACAudioPacket {
+        packet_type,
+        aac_data: &input[1..size],
```

Comment:
Shouldn't this be `1..(size - 1)`? Otherwise the last byte is part of this here and the input data we pass forward